### ISSUE TITLE
Liquidity NaN denominator

### ISSUE THREAD

**Issue by W3stside** [gnosis/dex-react#500]
Denominator values showing `NaN`  - likely from a `0` spread - needs investigation

![image](https://user-images.githubusercontent.com/21335563/73920074-0225a480-48c5-11ea-87b1-80758c3986de.png)

**Comment by anxolin** [comment]
@W3stside does it still reproduce in our current version?

**Comment by alfetopito** [comment]
I've disabled `0` spread on this PR https://github.com/gnosis/dex-react/pull/643
Although that won't prevent direct contract interaction.
Didn't dig deep though to check whether that's the root cause.

**Comment by ykplayer8** [comment]
I have similar issue
![image](https://user-images.githubusercontent.com/51191396/75858115-91f93880-5e08-11ea-8f86-451cac7eb46a.png)

**Comment by alfetopito** [comment]
@ykplayer8 could you share your address so we can look at the values in the contract and investigate?

If you rather not, could you tell what are the values you placed on those orders for sell/buy amounts?

**Comment by ykplayer8** [comment]
@alfetopito My address is 0x84443F61efc60D10DA9F9a2398980CD5748394BB
In most of my orders I press "Max" button, then change " , " to " . " other way I see "Invalid amount" then submitted order. Also I use GS + WC not metamask. 
![image](https://user-images.githubusercontent.com/51191396/75884141-59238880-5e35-11ea-944b-7d40cb5ff34c.png)

**Comment by alfetopito** [comment]
The prices in your orders (from the contract) look alright:

```
id: buy amount / sell amount = price
0: 19032801561808570507 (WETH)/19723110426744632650 (sETH) = 0.9649999999999999999873245145
1: 32500000000000000000 (sUSD)/50000000000000000000 (OWL) = 0.65
2: 534500000000000000000 (sUSD)/500000000000000000000 (DAI) = 1.069
3: 65000000000000000000 (sUSD)/100000000000000000000 (OWL) = 0.65
4: 641940000000000000000 (sUSD)/600000000000000000000 (DAI) = 1.0699
5: 651339707285067593251 (DAI)/664632354372517952297 (sUSD) = 0.9799999999999999999999097245
6: 2189400019946526048241 (sUSD)/2050000018676522517080 (DAI) = 1.067999999999999999999785366
7: 638047060197617234205 (DAI)/664632354372517952297 (sUSD) = 0.9599999999999999999998194491
8: 2162750019703731255519 (sUSD)/2050000018676522517080 (DAI) = 1.054999999999999999999804878
9: 631400736653892054682 (DAI)/664632354372517952297 (sUSD) = 0.9499999999999999999997743113
10: 1791623656278523922627 (DAI)/1885919638187919918555 (sUSD) = 0.9499999999999999999998674387
11: 2005466739809089518727 (sUSD)/1899116230879819620007 (DAI) = 1.055999999999999999999793588
12: 2701653335505344623210 (DAI)/2843845616321415392853 (sUSD) = 0.9499999999999999999998769272
13: 2885559421991860838297 (sUSD)/2709445466659024261312 (DAI) = 1.064999999999999999999896658
14: 2760665012418872645290 (DAI)/2907493430667585724371 (sUSD) = 0.9494999999999999999999090282
15: 2964694590794142339300 (sUSD)/2790300791335663378165 (DAI) = 1.062499999999999999999888005
16: 2840146727728269517769 (DAI)/2991202451530562946571 (sUSD) = 0.9494999999999999999999450054
17: 3050050480188260018056 (sUSD)/2870635746059538840524 (DAI) = 1.062499999999999999999738734
18: 1089062500000000000000 (sUSD)/1025000000000000000000 (DAI) = 1.0625
19: 5765236794664617251099 (sUSD)/5395635746059538840524 (DAI) = 1.068499999999999999999834311
20: 5512267841663078808695 (DAI)/5784121554735654573657 (sUSD) = 0.9529999999999999999999790807
21: 5877017465641090735434 (sUSD)/5546972596169033256663 (DAI) = 1.059499999999999999999919145
22: 5932487191602781068001 (sUSD)/5546972596169033256663 (DAI) = 1.069499999999999999999985848
23: 5896431869727682351832 (sUSD)/5546972596169033256663 (DAI) = 1.062999999999999999999861366
```

Will check the calculation in the UI.

**Comment by alfetopito** [comment]
Hmmmm, nothing apparent in the UI code.

I placed some orders (on Rinkeby) with the exact same amounts and prices are showing up fine.

```
187: 32500000000000000000 (DAI-7) / 50000000000000000000 (OWL-0) = 0.65
188: 32500000000000000000 (DAI-7) / 50000000000000000000 (OWL-0) = 0.65
```

![Screenshot from 2020-03-04 13-50-27](https://user-images.githubusercontent.com/43217/75902664-24a4d200-5e1f-11ea-8060-c74ee5b4c381.png)

**Comment by anxolin** [comment]
> In most of my orders I press "Max" button, then change " , " to " . " other way I see "Invalid amount" then submitted order. Also I use GS + WC not metamask. 

@ykplayer8  the "," and "." substitution seems to be a different problem. I'll create a new issue for that --> https://github.com/gnosis/dex-react/issues/691

Also, can you provide a bit more of info:
1. Which browser and wallet are you using?
2. Does it happen always, or some times? 
3. Can you open developer tools and show a screenshot of the console, so we can check if there's an error there that would give us a hint?

Thanks a lot for your feedback and your help!

**Comment by ykplayer8** [comment]
@anxolin 
1. Google chrome+Gnosis safe with wallet connect
2. Always
![image](https://user-images.githubusercontent.com/51191396/76247741-173d7c80-6251-11ea-9311-35b521fec78d.png)
![image](https://user-images.githubusercontent.com/51191396/76247849-4653ee00-6251-11ea-91d2-800ab4cc4864.png)
![image](https://user-images.githubusercontent.com/51191396/76247930-671c4380-6251-11ea-83ca-1d707242acfa.png)

**Comment by ykplayer8** [comment]
And after I swap "," to "." I am able to place order with mistakes running on console 
![image](https://user-images.githubusercontent.com/51191396/76248480-75b72a80-6252-11ea-9be0-1e1a91f52dfe.png)

**Comment by alfetopito** [comment]
Getting some similar results for prod version using Chrome and Gnosis Safe (via Wallet Connect).

### 1
Right after connecting, getting the `MaxListenersExceededWarning` warning

![Screenshot from 2020-03-09 13-14-32](https://user-images.githubusercontent.com/43217/76253555-20eec200-6208-11ea-8c7e-940b64e3d4ba.png)

### 2
And once in awhile, the following errors (I would guess during some block updates):

![Screenshot from 2020-03-09 13-14-58](https://user-images.githubusercontent.com/43217/76253586-306e0b00-6208-11ea-9fe2-66ad66cfb131.png)

----
On staging, `1` also happens on connection, and `2` from time to time.

I also observed a different error

### 3
```
Access to XMLHttpRequest at 'https://mainnet.infura.io/v3/8b4d9b4…' from origin 'https://dex.staging.gnosisdev.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

useTokenBalances.ts:78 [useTokenBalances] Error for 
{id: 8, name: "Synth sETH", symbol: "sETH", decimals: 18, address: "0x5e74C9036fb86BD7eCdcb084a0673EFc32eA31cb", …}
 0x49d56a0ae3608b683755cd54e2020134319213e5 0x6F400810b62df8E13fded51bE75fF5393eaa841F Error: Invalid JSON RPC response: {"id":1126,"jsonrpc":"2.0"}
    at Object.InvalidResponse (errors.js:42)
    at index.js:144
    at index.js:139
    at index.js:14
    at p (index.js:47)
    at d (index.js:47)
    at index.js:14
    at index.js:14
    at setImmediate.js:46
    at d (setImmediate.js:69)
```

![Screenshot from 2020-03-09 13-22-52](https://user-images.githubusercontent.com/43217/76254107-1c76d900-6209-11ea-95ae-576416117bb3.png)

**Comment by alfetopito** [comment]
Seeing a case of [heizenbug](https://en.wikipedia.org/wiki/Heisenbug).

When debug is on (`window.toogleDebug()`), the issue `2` rarely occurs. When debug is off, it happens way more often.

This is something wallet connect related. Doesn't seem to be related to the NaN issue though. Will create a separated issue.

The same for prod, staging and a local dev server.

**Comment by alfetopito** [comment]
@ykplayer8 I see at the very top in your console logs a message about the page not being whitelisted in some extension.
Can you try to open the page in incognito mode and let us know if you still see the issue? 
And also if possible share which extensions are active, as well as what's your locale.

**Comment by ykplayer8** [comment]
@alfetopito In incognito mode it looks better
![image](https://user-images.githubusercontent.com/51191396/76338866-60053c00-630a-11ea-9265-e67232547ee5.png)

**Comment by ykplayer8** [comment]
![image](https://user-images.githubusercontent.com/51191396/76339007-9c389c80-630a-11ea-9c4a-bcaed8b03c16.png)

**Comment by ykplayer8** [comment]
What is "locale" mean?

**Comment by ykplayer8** [comment]
And I want mentioned one more. One I interecating  with dfusion with Gnosis Safe it is possible to interact when whitelisted toogle button in switch off mode. Usually user must switch on this button before interaction.
![image](https://user-images.githubusercontent.com/51191396/76342069-00f5f600-630f-11ea-93ef-2b015889e135.png)

**Comment by alfetopito** [comment]
> @alfetopito In incognito mode it looks better but not ideal
> ![image](https://user-images.githubusercontent.com/51191396/76339312-f20d4480-630a-11ea-941e-f24c303857e8.png)

Yes, looks much better. I've created another issue to deal with that specific error coming from WalletConnect, and we had in another project the warning about max listeners. It's likely not a memory, but instead just the need to increase the amount of listeners.

For now, I'll try to reproduce using the same extensions as you have.

The important question is, do you see the `NaN` in your orders when on incognito?

**Comment by ykplayer8** [comment]
Yes, I see NaN in my orders when incognito

**Comment by alfetopito** [comment]
> And I want mentioned one more. One I interecating with dfusion with Gnosis Safe it is possible to interact when whitelisted toogle button in switch off mode. Usually user must switch on this button before interaction.
> ![image](https://user-images.githubusercontent.com/51191396/76342069-00f5f600-630f-11ea-93ef-2b015889e135.png)

With regards to the safe extension, just checked with the Safe people.
When using WalletConnect, the extension is 'ignored'. That's because the WalletConnect web3provider is used instead of the one provided by the Safe extension. Thus, that toggle has no effect.

**Comment by alfetopito** [comment]
YEEEEESSS I can reproduce it!!!

Switched my system locale fully to a language that uses comma as a decimal separator and now I see it (changing only parts of the locale did not work).

![Captura de tela de 2020-03-10 14-24-57](https://user-images.githubusercontent.com/43217/76360818-fae62200-62da-11ea-85a4-bc8145837ef6.png)
